x-pack/legacy/plugins/task_manager/task_store.ts

```diff
@@ -177,6 +178,8 @@ export class TaskStore {
                 taskType: { type: 'keyword' },
                 scheduledAt: { type: 'date' },
                 runAt: { type: 'date' },
+                startedAt: { type: 'date' },
+                retryAt: { type: 'date' },
```

Comment:
Task Manager has logic to run `indices.putTemplate` if the existing template belongs to an older version, so that's good. A new `.kibana_task_manager` index will be created with the correct mappings.

However, if there is an existing `.kibana_task_manager` index, we'll need a migration for it because old mappings are baked into that index.

Reply:
Is there a migration mechanism at this time? From what I see, task manager uses templates which don't reflect changes to created indices.

Reply:
The template management in task manager is there so that a template installed by an older version can be upgraded. The benefit of having a template is pretty specific: if a user decided to delete the `.kibana_task_manager` while Kibana is running, new tasks that get indexed will have the correct mapping. Other parts of Kibana don't do this: things go really bad if you delete other system indices that Kibana manages while Kibana is running.